src/actions/commands/fold.ts

```diff
@@ -72,3 +75,42 @@ class CommandOpenAllFoldsRecursively extends CommandFold {
   keys = ['z', 'O'];
   commandName = 'editor.unfoldRecursively';
 }
+
+@RegisterAction
+class AddFold extends BaseOperator {
+  override modes = [Mode.Normal, Mode.Visual];
+  keys = ['z', 'f'];
+
+  readonly commandName = 'editor.createFoldingRangeFromSelection';
+
+  public async run(vimState: VimState, start: Position, end: Position): Promise<void> {
+    const previousSelections = vimState.lastVisualSelection;  // keep in case of Normal mode
+    vimState.editor.selection = new vscode.Selection(start, end);
+    await vscode.commands.executeCommand(this.commandName);
+    vimState.lastVisualSelection = previousSelections;
+    vimState.cursors = [new Cursor(start, start)];
+    await vimState.setCurrentMode(Mode.Normal);  // Vim behavior
+  }
+}
+
+@RegisterAction
+class RemoveFold extends BaseCommand {
+  override modes = [Mode.Normal, Mode.Visual];
+  keys = ['z', 'd'];
+  readonly commandName = 'editor.removeManualFoldingRanges';
+
+  override async exec(position: Position, vimState: VimState): Promise<void> {
+
+    const selectionForFold = vimState.currentMode === Mode.Visual
+      ? vimState.editor.selection
+      : new vscode.Selection(position, position);
+
+    const previousSelections = vimState.lastVisualSelection;  // keep in case of Normal mode
+    vimState.editor.selection = selectionForFold;
```

Comment:
Ditto - shouldn't the editor's selection be correct already?

Reply:
This is needed for the normal mode case.

Reply:
Why's that? Shouldn't `vimState.editor.selection` be `(position, position)` already, in Normal mode?

Reply:
I'll check it.

Reply:
You're correct. Updated.